singer_sdk/connectors/sql.py

```diff
@@ -477,9 +477,16 @@ def table_exists(self, full_table_name: str) -> bool:
         Returns:
             True if table exists, False if not, None if unsure or undetectable.
         """
+        # pftn -> Parsed Full Table Name [0] = db, [1] = schema, [2] = table
+        pftn: tuple = SQLConnector.parse_full_table_name(
+            self, full_table_name=full_table_name
+        )
+        table_name: str = pftn[2]
+        schema_name: str = pftn[1]
+
         return cast(
             bool,
-            sqlalchemy.inspect(self._engine).has_table(full_table_name),
+            sqlalchemy.inspect(self._engine).has_table(table_name, schema_name),
```

Comment:
I think for the most part, we usually expect a single DB as source.

However, if the table in question does reference something other than the default db (e.g. if `full_table_name` is in `db.schema.table` format), we may want to do something here to handle db_name as well. That said, I don't know what the generic way to do this would be, and probably we can leave omitted if the parsing does not return an explicit db name.

What do you think?

Reply:
The only inputs has_table() takes is table name and scheme as far as I can tell.

https://docs.sqlalchemy.org/en/14/core/reflection.html#sqlalchemy.engine.reflection.Inspector.has_table

Reply:
@BuzzCutNorman yeah, the db is set at the connection/engine level 👍

Reply:
Though in a target instance I think I would like it to only look at the database I specified in the connection.  which like you said should be a single database.  I know you can have fall back connections to different servers, but I haven't seen different databases given in SQLA examples.

Reply:
@BuzzCutNorman + @edgarrmondragon - Yeah, makes sense. 👍